assets/blocks/reader-registration/view.js

```diff
@@ -53,5 +65,45 @@ import './style.scss';
 				res.json().then( ( { message, data } ) => endLoginFlow( message, res.status, data ) );
 			} );
 		} );
+
+		const googleLogin = container.querySelector( '#newspack-google-login' );
+		if ( googleLogin ) {
+			googleLogin.onclick = () => {
+				startLoginFlow();
+				const checkLoginStatus = () => {
+					fetch( '/wp-json/newspack/v1/login/google/register', {
+						headers: new Headers( {
+							'X-WP-Nonce': window.newspack_reader_activation_data.nonce,
+						} ),
+					} ).then( res => {
+						res.json().then( ( { message, data } ) => endLoginFlow( message, res.status, data ) );
+					} );
+				};
+				fetch( '/wp-json/newspack/v1/login/google' )
+					.then( res => res.json().then( data => Promise.resolve( { data, status: res.status } ) ) )
+					.then( ( { data, status } ) => {
+						if ( status !== 200 ) {
+							endLoginFlow( data.message, status );
+						} else {
+							const authWindow = window.open(
+								'about:blank',
+								'newspack_google_login',
+								'width=500,height=600'
+							);
+							if ( authWindow ) {
+								authWindow.location = data;
+								const interval = setInterval( () => {
+									if ( authWindow.closed ) {
+										checkLoginStatus();
+										clearInterval( interval );
+									}
+								}, 500 );
+							} else {
+								endLoginFlow();
+							}
+						}
+					} );
+			};
+		}
 	} );
 } )( window.newspackReaderActivation );
```

Comment:
I think there's a race condition happening here. Sometimes when I load the page, `window.newspackReaderActivation` is `undefined`, which you can confirm by console.logging the value inside this file's IFEE closure. When this happens, the flow is unavailable and nothing happens when you click the "Sign in with Google" button.

Reply:
More details in [comment below](https://github.com/Automattic/newspack-plugin/pull/1781#issuecomment-1192719034).